Add support for building Android x86 dependencies

GeorgeK1ng opened the pull request:
Added minimal changes to support Android x86. 

Opus fix were added, because all builds fails currently without that. Have zero experience in this area, but managed to make it compile dependencies again.

GeorgeK1ng committed:
Android x86 profiles

GeorgeK1ng committed:
Add Android x86 dependencies build support

GeorgeK1ng committed:
opus vs ffmpeg fix

GeorgeK1ng committed:
Force shared build on Android x86

kambala-decapitator commented:
what is the use case? So far nobody needed that

kambala-decapitator commented:
it seems that you're the only person who needs that :) in such case I'd suggest to build dependencies and VCMI on your own (on local machine, via GitHub Actions, etc.) without adding the support burden to the project.

if you need help with testing on a 32-bit ARM android, just ask.

GeorgeK1ng commented:
What's problem with adding Android x86 support for VCMI? You don't need to do it yourself, it's already done in PR.

kambala-decapitator commented:
maintenance burden for a platform that nobody uses. If its build starts failing in the future, who will investigate that? Do you wish to be Android x86 maintainer?

adding profiles to the repo is OK of course.

GeorgeK1ng commented:
It's still a solution for old x86 machines for browsing internet.

GeorgeK1ng committed:
Merge branch 'vcmi:main' into main

GeorgeK1ng committed:
Update Android x86 profile settings and API level

GeorgeK1ng committed:
Modify Android x86 profile settings and options

Updated Android x86 profile settings and added configuration for file functions macros.

GeorgeK1ng committed:
Update rebuildDependencies.yml workflow file

GeorgeK1ng committed:
Update rebuildDependencies.yml workflow file

GeorgeK1ng committed:
Update opus version conflict resolution

Commented out the version conflict fix for opus.

GeorgeK1ng committed:
Update common

GeorgeK1ng committed:
Merge branch 'vcmi:main' into main

GeorgeK1ng committed:
Update common

GeorgeK1ng commented:
Seems Android x86 dependencies are builded just fine. Even after latest changes for new AI. Can we merge it, so we can support it until it dies please?

GeorgeK1ng committed:
Create linux.sh

GeorgeK1ng committed:
Merge branch 'main' into main

GeorgeK1ng committed:
Fix FFmpeg

GeorgeK1ng commented:
Can we please merge with squash this PR? It currently just adds Android x86 profile which builds fine.

I don't know when new [Bliss OS](https://blissos.org/) commes out, but I would like to add support for VCMI for their Android x86 port.

GeorgeK1ng committed:
ReRun CI
